scipy/signal/_peak_finding_utils.pyx

```diff
@@ -46,7 +46,7 @@ def _argmaxima1d(np.float64_t[:] x not None):
     """
     # Preallocate, there can't be more maxima than half the size of `x`
     cdef np.ndarray[np.int64_t, ndim=1] maxima
-    maxima = np.empty(x.shape[0] // 2, dtype=np.int64)
+    maxima = np.empty(x.shape[0] // 2, dtype=np.intp)
```

Comment:
Why prefer `intp` (platform-specific?) over `int64` here?

Reply:
It's the numpy equivalent of `Py_ssize_t`, and is the standard type of any index returning function in numpy: will not waste memory on 32 bit systems, and will work without overflow risk when 128 bit systems take over!

Reply:
In that case, shouldn't the declaration in the line above have the same dtype `np.intp_t`? Maybe the build failures are related to that? 
> `ValueError: Buffer dtype mismatch, expected 'int64_t' but got 'int'`

Reply:
Yes indeed, have submitted a new commit fixing that.